include/ozo/libpq.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <ozo/type_traits.h>
+
+#include <libpq-fe.h>
+
+#include <boost/iterator/iterator_facade.hpp>
+#include <memory>
+#include <vector>
+
+
+namespace ozo::libpq {
+
+class value {
+public:
+    value(PGresult* r, int row, int col) : r_(r), row_(row), col_(col) {}
+
+    inline oid_t oid() const { return PQftype(r_, col_); }
+    inline const char* bytes() const { return PQgetvalue(r_, row_, col_); }
+    inline int size() const { return PQgetlength(r_, row_, col_); }
+
+private:
+    PGresult* r_;
+    int row_;
+    int col_;
+};
+
+
+class row {
+public:
+    class iterator : public boost::iterator_facade<
+        iterator,
+        value,
+        boost::forward_traversal_tag, // maybe will implement random access later
+        const value&,
+        int
+    > {
+    public:
+        iterator() = default;
+        iterator(PGresult* r, int row, int col) : r_(r), row_(row), col_(col),
+            v_(r, row, col) {}
+
+    private:
+        const value& dereference() const {
+            return v_;
+        }
+
+        bool equal(const iterator& rhs) const {
+            return r_ == rhs.r_ && row_ == rhs.row_ && col_ == rhs.col_;
+        }
+
+        void increment() {
+            if (col_ + 1 < PQnfields(r_)) {
+                ++ col_;
+                v_ = {r_, row_, col_};
+            } else {
+                *this = {};
+            }
+        }
+
+        PGresult* r_ {nullptr};
+        int row_ {0};
+        int col_ {0};
+
+        value v_ {nullptr, 0, 0};
```

Comment:
The value contains same data elements as iterator itself - it is very strange and redundant.